### PR TITLE
Fix: Memory Issues

### PR DESCRIPTION
**Pull request by tomiir**
## Changelog

- Install controller 0.23.0 with trimmed down state.
- Virtualized nfts to fix issue for huge number of tokens in collections.
- Fixes issue with ICNS as an NFT collection
- Updates code to 0.6.0 base

### Type of changes included:

- [ ] Components
- [X] Business Logic
- [X] Bug fixes
- [ ] Styling
- [ ] Code Refactor

### Tested on:

- [X] Chrome
- [ ] Firefox
- [ ] Safari

### CODE REVIEW

File: source/Background/Keyring/index.js [hunk 1]
```diff
@@ -206,22 +206,12 @@ export const getKeyringHandler = (type, keyring) => ({
...
-        || Object.values(assets)?.some((asset) => asset.amount === 'Error')
-        || refresh;
-      if (shouldUpdate) {
-        assets = await keyring.getBalances();
-      } else {
-        keyring.getBalances();
-      }
-      assets = parseAssetsAmount(assets);
-      return (assets || []).map((asset) => recursiveParseBigint(asset));
+      const assets = await keyring.getBalances();
```
Comment (angarita-dev) [inline source/Background/Keyring/index.js:227]: 👍

**Review by angarita-dev — APPROVED** [review]

File: source/Background/Keyring/index.js [hunk 2]
```diff
@@ -339,11 +329,7 @@ export const getKeyringHandler = (type, keyring) => ({
...
   [HANDLER_TYPES.GET_NFTS]: async ({ refresh = false }) => {
-    const { wallets, currentWalletId } = await keyring.getState();
-    let collections = wallets?.[currentWalletId]?.collections || [];
-    if (!collections.length || refresh) {
-      collections = await keyring.getNFTs({ subaccount: currentWalletId, refresh });
-    }
+    const collections = await keyring.getNFTs({ refresh });
     return (collections || [])?.map((collection) => recursiveParseBigint(collection));
   },
   [HANDLER_TYPES.TRANSFER_NFT]: async ({ to, nft }) => {
```
Comment (ignaciosantise) [inline source/Background/Keyring/index.js:352]: `transferNFT` now returns a boolean, you need to change this

Comment (ignaciosantise) [reply source/Background/Keyring/index.js:352]: `registerToken` also changed its response

Comment (ignaciosantise) [reply source/Background/Keyring/index.js:352]: And i think `registerNFT` too

**Comment by ignaciosantise** [comment]
also update the controller to 0.23.2

File: source/components/Profile/index.jsx [hunk 3]
```diff
@@ -114,9 +114,9 @@ const Profile = ({ disableProfile }) => {
   };
 
   const executeAccountSwitch = (wallet) => {
-    dispatch(setCollections({ collections: [], principalId }));
     sendMessage({ type: HANDLER_TYPES.SET_CURRENT_PRINCIPAL, params: wallet },
       (state) => {
+        console.log('state on account switch from profile', state);
```
Comment (ignaciosantise) [inline source/components/Profile/index.jsx]:
```suggestion
```

File: source/views/Extension/Views/SendNFT/components/NFTDisplay/index.jsx [hunk 4]
```diff
@@ -10,7 +10,7 @@ const NFTDisplay = ({ nft }) => {
   const classes = useStyles();
 
   const nftDefaultTag = NFT_COLLECTION_DEFAULT_TYPES[nft?.canisterId];
-
+  console.log('nft to display', nft);
```
Comment (ignaciosantise) [inline source/views/Extension/Views/SendNFT/components/NFTDisplay/index.jsx]:
```suggestion
```

File: source/views/Extension/Views/WalletDetails/index.jsx [hunk 5]
```diff
@@ -246,9 +246,9 @@ const WalletDetails = () => {
   };
 
   const executeAccountSwitch = (wallet) => {
-    dispatch(setCollections({ collections: [], principalId }));
     sendMessage({ type: HANDLER_TYPES.SET_CURRENT_PRINCIPAL, params: accountSwitchId },
       (state) => {
+        console.log('state on account switch from details', state);
```
Comment (ignaciosantise) [inline source/views/Extension/Views/WalletDetails/index.jsx]:
```suggestion
```

**Review by ignaciosantise — APPROVED** [review]
🚀

**Pull request merged**